Source/WebCore/Modules/reporting/Report.h

```diff
@@ -25,25 +25,33 @@
 
 #pragma once
 
-#include <wtf/JSONValues.h>
+#include "COEPInheritenceViolationReportBody.h"
+#include "CORPViolationReportBody.h"
+#include "CSPViolationReportBody.h"
+#include "DeprecationReportBody.h"
 #include "ReportBody.h"
+#include "TestReportBody.h"
+
+#include <wtf/JSONValues.h>
 #include <wtf/RefCounted.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
 class FormData;
 
+using ReportBodyType = std::variant<Ref<CSPViolationReportBody>, Ref<COEPInheritenceViolationReportBody>, Ref<CORPViolationReportBody>, Ref<DeprecationReportBody>, Ref<TestReportBody>>;
```

Comment:
This shouldn't be changed in C++, but something like this will be needed in SerializedTypeInfo.cpp.  Do something like WebCore::TimingFunction, then in generate_serialized_type_info pretend ReportBody contains a variant of its subclasses.  We currently just skip it with the continue after "if type.members_are_subclasses:"

Reply:
It looks like we can change it from a RefPtr to a Ref, though.

Reply:
Thanks. I didn't know about "subclasses". Makes it wayyy cleaner.

What do you think can change from RefPtr to Ref? Report::m_body?

Reply:
probably, but can be done in a separate PR.